exonum/benches/criterion/crypto.rs

```diff
@@ -0,0 +1,59 @@
+// Copyright 2018 The Exonum Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use criterion::{AxisScale, Bencher, Criterion, ParameterizedBenchmark, PlotConfiguration,
+                Throughput};
+use exonum::crypto::{gen_keypair, hash, sign, verify};
+use num::pow::pow;
+
+fn bench_sign(b: &mut Bencher, &count: &usize) {
+    let (_, secret_key) = gen_keypair();
+    let data = (0..count).map(|x| (x % 255) as u8).collect::<Vec<u8>>();
+    b.iter(|| sign(&data, &secret_key))
+}
+
+fn bench_verify(b: &mut Bencher, &count: &usize) {
+    let (public_key, secret_key) = gen_keypair();
+    let data = (0..count).map(|x| (x % 255) as u8).collect::<Vec<u8>>();
+    let signature = sign(&data, &secret_key);
+    b.iter(|| verify(&signature, &data, &public_key))
+}
+
+fn bench_hash(b: &mut Bencher, &count: &usize) {
+    let data = (0..count).map(|x| (x % 255) as u8).collect::<Vec<u8>>();
+    b.iter(|| hash(&data))
+}
+
+pub fn bench_crypto(c: &mut Criterion) {
+    ::exonum::crypto::init();
+
+    c.bench(
+        "hash",
+        ParameterizedBenchmark::new("hash", bench_hash, (6..16).map(|i| pow(2, i)))
```

Comment:
Nitpick: It's not obvious what `(6..16).map(|i| pow(2, i))` stands for.

Reply:
It's an argument for test, i am not sure is it should be obvious. We interesting about speed of methods with argument in between some size.

64 - is relatively small message
65536 - is relatively big

Reply:
Yep, I got that (after reading `critertion` source code mainly). What I meant is that it would be nice to add explaining remark to sources. I guess, it will make work with those lines in 2-3 months less confusing.